bin/deploy_automation/_shared_functions.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+# Traps error codes to provide better error context
+_trap_err() {
+  declare exit_code=$?
+  declare fn=( "${FUNCNAME[@]:1}" )
+  if [ "$exit_code" != 0 ] && [ "$exit_code" != 99 ]; then
+    >&2 printf "\n${RED}ERROR: failed to execute command \"%s\". Exit code %d. Invoked from %s.${RESET}\n" "$BASH_COMMAND" "${exit_code}" "${fn[*]} $0"
+  fi
+  exit $exit_code
+}
+trap '_trap_err' EXIT
+
+_exit_with_err_msg() {
+  >&2 printf "\n\n${CYAN}[%s]: ${BOLD}${RED}ERROR: %s${RESET}\n\n" "$(date +'%Y-%m-%dT%H:%M:%S%z')" "$*"
+  exit 99
+}
+
+_log() {
+  >&2 printf "${CYAN}[%s]: ${BOLD}${GREEN}%s${RESET}\n" "$(date +'%Y-%m-%dT%H:%M:%S%z')" "$*"
+}
+
+_trace() {
+  >&2 printf "${RESET}%s\n" "$*"
+}
+
+_get_latest_commit() {
+  declare branch_or_tag_name="$1"
+  git log -n 1 --pretty=format:"%h" "$branch_or_tag_name"
+}
+
+_fetch_current_release_checklist_from_github() {
+  # Fetch current release checklist using github API and prints a json object.
+  # If there is no checklist available it will print json object "null" (without quotes)
+  declare url="${GITHUB_REPOSITORY_API}/pulls?state=open"
+  _trace "Fetching [RELEASE_CHECKLIST] draft pull request from repo $GITHUB_REPOSITORY_URL/pulls"
+  http --ignore-stdin --check-status --timeout=30 \
+      GET "$url" "Authorization:token $GITHUB_TOKEN" \
+    | jq 'map(select(.title | test("^\\[RELEASE_CHECKLIST\\] ")))[0]'
```

Comment:
is this still supposed to say `test("`?

Reply:
Yes, this is `jq` syntax for a `where` like clause...

More info here: https://stedolan.github.io/jq/manual/#test(val),test(regex;flags)